cmd/bucket-handlers-listobjects.go

```diff
@@ -104,31 +77,34 @@ func (api objectAPIHandlers) ListObjectsV2Handler(w http.ResponseWriter, r *http
 		// Then we need to use 'start-after' as marker instead.
 		marker = startAfter
 	}
+
 	// Validate the query params before beginning to serve the request.
 	// fetch-owner is not validated since it is a boolean
 	if s3Error := validateListObjectsArgs(prefix, marker, delimiter, maxKeys); s3Error != ErrNone {
-		// return empty response if invalid marker
-		//TODO: avoid this pattern when moving to errors package
-		if s3Error == ErrInvalidPrefixMarker {
+		writeErrorResponse(w, s3Error, r.URL)
+		return
+	}
+	// Marker is set validate pre-condition.
+	if !api.gateway && marker != "" {
```

Comment:
you can remove this code:
```
        // Marker is set validate pre-condition.
        if !api.gateway && marker != "" {
                // Marker not common with prefix is not implemented.Send an empty response
                if !hasPrefix(marker, prefix) {
                        listObjectsInfo := ListObjectsInfo{}
                        response := generateListObjectsV2Response(bucket, prefix, token, marker, startAfter, delimiter, fetchOwner, listObjectsInfo.IsTruncated, maxKeys, listObjectsInfo.Objects, listObjectsInfo.Prefixes)
                        writeSuccessResponseXML(w, encodeResponse(response))
                        return
                }
        }
```

and let the underlying object-layer handle this however it wants. i.e fs and xl can do a hasPrefix() check and other implementations need not.

Reply:
done